resources/configure.rb

```diff
@@ -0,0 +1,230 @@
+    -Djava.io.tmpdir=${ES_TMPDIR}
+    -XX:+HeapDumpOnOutOfMemoryError
+    9-:-XX:+ExitOnOutOfMemoryError
+    -XX:ErrorFile=/var/log/elasticsearch/hs_err_pid%p.log
+    8:-XX:+PrintGCDetails
+    8:-XX:+PrintGCDateStamps
+    8:-XX:+PrintTenuringDistribution
+    8:-XX:+PrintGCApplicationStoppedTime
+    8:-Xloggc:/var/log/elasticsearch/gc.log
+    8:-XX:+UseGCLogFileRotation
+    8:-XX:NumberOfGCLogFiles=32
+    8:-XX:GCLogFileSize=64m
+    9-:-Xlog:gc*,gc+age=trace,safepoint:file=/var/log/elasticsearch/gc.log:utctime,pid,tags:filecount=32,filesize=64m
+  ).freeze
+
+# These are the default settings. Most of the time, you want to override
+# the `configuration` attribute below. If you do override the defaults, you
+# must supply ALL needed defaults, and don't use nil as a value in the hash.
+property :default_configuration, Hash, default: {
+  'cluster.name' => 'elasticsearch',
+  'node.name' => Chef::Config[:node_name],
+}
+
+# These settings are merged with the `default_configuration` attribute,
+# allowing you to override and set specific settings. Unless you intend to
+# wipe out all default settings, your configuration items should go here.
+#
+property :configuration, Hash, default: {}
+
+include ElasticsearchCookbook::Helpers
+
+action :manage do
+  # lookup existing ES resources
+  es_user = find_es_resource(Chef.run_context, :elasticsearch_user, new_resource)
+  es_svc = find_es_resource(Chef.run_context, :elasticsearch_service, new_resource)
+  es_install = find_es_resource(Chef.run_context, :elasticsearch_install, new_resource)
+
+  default_configuration = new_resource.default_configuration.dup
+  # if a subdir parameter is missing but dir is set, infer the subdir name
+  # then go and be sure it's also set in the YML hash if it wasn't given there
+  if new_resource.path_data && default_configuration['path.data'].nil?
+    default_configuration['path.data'] = new_resource.path_data
+  end
+
+  if new_resource.path_logs && default_configuration['path.logs'].nil?
+    default_configuration['path.logs'] = new_resource.path_logs
+  end
+
+  # Calculation for memory allocation; 50% or 31g, whatever is smaller
+  #
+  unless new_resource.allocated_memory
+    half = ((node['memory']['total'].to_i * 0.5).floor / 1024)
+    malloc_str = (half > 30_500 ? '30500m' : "#{half}m")
+    new_resource.allocated_memory malloc_str
+  end
+
+  # Create ES directories
+  #
+  [new_resource.path_conf, "#{new_resource.path_conf}/scripts"].each do |path|
+    directory path do
+      owner es_user.username
+      group es_user.groupname
+      mode '0750'
+      recursive true
+      action :create
+    end
+  end
+
+  directory new_resource.path_data do
+    owner es_user.username
+    group es_user.groupname
+    mode '0750'
+    recursive true
+    action :create
+  end
+
+  if new_resource.path_data.is_a?(String)
+    directory new_resource.path_data do
+      owner es_user.username
+      group es_user.groupname
+      mode '0755'
```

Comment:
Shouldn't it be `750`?

Reply:
If you've got documentation that it should be. Happy for a PR 👍🏼

Reply:
to be more exact during a chef run:
```
    * directory[/data] action create
      - change mode from '0755' to '0750'
    * directory[/data] action create
      - change mode from '0750' to '0755'
```

Reply:
Ace, if you can make the PR, I'l merge it today 👍🏼

Reply:
You are setting 750 in the above resource, and 755 here.

Reply:
https://github.com/sous-chefs/elasticsearch/pull/781